### PR TITLE
Use Joi to validate a loaded gist

### PR DESCRIPTION
**Pull request by pdehaan**
Async validation of a loaded Gist using Joi.

### CODE REVIEW

**Comment by pdehaan** [comment]
r? @darkwing 

It ain't pretty, but it's also in this repo, so you're free to edit before merging (or closing with impunity and a whimsical r- GIF &mdash; which is always hilarious!)

File: client/static/assets/scripts/controllers.js [hunk 1]
```diff
@@ -16,20 +16,38 @@ angular.module('LoadsApp')
...
 
       var gistId = jQuery('#gist').val();
       var $resultDestination = jQuery('.result-destination');
 
-      console.log($resultDestination.get(0));
-
       $resultDestination.removeClass('error').removeClass('valid');
       jQuery.getJSON('/api/gist/' + gistId).done(function(data) {
-        console.log('done!', data.success, data);
+        // console.log('done!', data.success, JSON.stringify(data.files[0].content, null, 2));
```
Comment (darkwing) [inline client/static/assets/scripts/controllers.js]: Since you'll need to rebase this PR, can you get rid of my missed console call?  *NOOB ALERT*

**Comment by pdehaan** [comment]
@darkwing Rebased and looks mergeable. Feel free to retest, or blindly merge or whatever.
It still uses `alert()` (eww, gross), but is a good enough starting point for now.

**Comment by darkwing** [comment]
Looks good!

**Pull request merged**